File: backend/src/main/java/com/lifeos/observability/metrics/dto/AppMetricsDto.java
package com.lifeos.observability.metrics.dto;

import lombok.*;

import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AppMetricsDto {

    private Instant timestamp;

    private Double heapUsedMb;

    private Double heapTotalMb;

    private Double heapMaxMb;

    private Long lastScanDurationMs;

    private Integer sessionsFoundInScan;

    private Integer pendingSessionsCount;

    private Integer pendingLogsCount;

    private Integer batteryLevel;

    private Boolean isCharging;

    private Boolean isWifiConnected;
}

File: android/app/src/main/java/com/karthik/lifeos/tracker/sync/dto/AppMetricsDto.kt
package com.karthik.lifeos.tracker.sync.dto

import java.time.Instant

/**
 * DTO matching the backend's AppMetricsDto.
 */
data class AppMetricsDto(
    val timestamp: String,     // ISO-8601 Instant string
    val heapUsedMb: Double,
    val heapTotalMb: Double,
    val heapMaxMb: Double,
    val lastScanDurationMs: Long,
    val sessionsFoundInScan: Int,
    val pendingSessionsCount: Int,
    val pendingLogsCount: Int,
    val batteryLevel: Int,
    val isCharging: Boolean,
    val isWifiConnected: Boolean
) {
    companion object {
        fun fromEpochMillis(millis: Long): String {
            return Instant.ofEpochMilli(millis).toString()
        }
    }
}
